doxyqml/qmlclass.py

```diff
@@ -109,8 +116,23 @@ def _export_footer(self, lst):
 
     def _export_content(self, lst):
         self._export_header(lst)
+        
+        # Public members.
         self._start_class(lst)
-        self._export_elements(lst)
+        
+        # Doxygen always adds Q_PROPERTY and Q_SIGNALS items as public
+        # members.
+        self._export_elements(lst, filter=lambda x:
+                              not isinstance(x, QmlAttribute) and
+                              not isinstance(x, QmlComponent))
+
+        # Private members.
+        if self.has_private_members:
+            lst.append("private:")
+            self._export_elements(lst, filter=lambda x:
+                                isinstance(x, QmlAttribute) or
+                                isinstance(x, QmlComponent))
```

Comment:
This looks good overall, but I wonder why is the condition to decide if an element is public is a bit different between here and the code in `add_element`? Is there a reason for checking for component id in `add_element` and not in `_export_content`?

If there is no reason, maybe you can factorize this distinction in a `is_public_element()` helper function, which could be used in the two `_export_elements()` calls and in `add_element()`?

Reply:
The additions to `add_element` was to avoid iterating through the entire element to determine if there were private elements.

I've implementing `is_public_element`, and modified `_export_elements` to accept any list of elements. This should avoid extra looping while keeping the heavy lifting confined to ` _export_content`.